java/core/src/test/java/com/lancedb/lance/TransactionTest.java

```diff
@@ -186,4 +190,98 @@ void testOverwrite(@TempDir Path tempDir) throws Exception {
+      try (Dataset updatedDataset = transaction.commit()) {
+        assertEquals(2, updatedDataset.version());
+        assertEquals("value1", updatedDataset.getConfig().get("key1"));
+        assertEquals("value2", updatedDataset.getConfig().get("key2"));
+
+        // Test 2: Delete configuration keys using deleteKeys
+        List<String> deleteKeys = Collections.singletonList("key1");
+        transaction =
+            updatedDataset
+                .newTransactionBuilder()
+                .operation(UpdateConfig.builder().deleteKeys(deleteKeys).build())
+                .build();
+        try (Dataset updatedDataset2 = transaction.commit()) {
+          assertEquals(3, updatedDataset2.version());
+          assertNull(updatedDataset2.getConfig().get("key1"));
+          assertEquals("value2", updatedDataset2.getConfig().get("key2"));
+
+          // Test 3: Update schema metadata using schemaMetadata
+          Map<String, String> schemaMetadata = new HashMap<>();
+          schemaMetadata.put("schema_key1", "schema_value1");
+          schemaMetadata.put("schema_key2", "schema_value2");
+
+          transaction =
+              updatedDataset2
+                  .newTransactionBuilder()
+                  .operation(UpdateConfig.builder().schemaMetadata(schemaMetadata).build())
+                  .build();
+          try (Dataset updatedDataset3 = transaction.commit()) {
+            assertEquals(4, updatedDataset3.version());
+            assertEquals(
+                "schema_value1", updatedDataset3.getLanceSchema().metadata().get("schema_key1"));
+            assertEquals(
+                "schema_value2", updatedDataset3.getLanceSchema().metadata().get("schema_key2"));
+
+            // Test 4: Update field metadata using fieldMetadata
+            Map<Integer, Map<String, String>> fieldMetadata = generateTestFieldConfigMap();
+
+            transaction =
+                updatedDataset3
+                    .newTransactionBuilder()
+                    .operation(UpdateConfig.builder().fieldMetadata(fieldMetadata).build())
+                    .build();
+            try (Dataset updatedDataset4 = transaction.commit()) {
+              assertEquals(5, updatedDataset4.version());
+
+              // Verify field metadata for field 0
+              Map<String, String> fieldMetadata0 =
+                  updatedDataset4.getLanceSchema().fields().get(0).getMetadata();
+              assertEquals("field0_value1", fieldMetadata0.get("field0_key1"));
+
+              // Verify field metadata for field 1
+              Map<String, String> field1Result =
+                  updatedDataset4.getLanceSchema().fields().get(1).getMetadata();
+              assertEquals("field1_value1", field1Result.get("field1_key1"));
+              assertEquals("field1_value2", field1Result.get("field1_key2"));
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private static Map<Integer, Map<String, String>> generateTestFieldConfigMap() {
```

Comment:
looks like this is used only once, why a separated method?

Reply:
<img width="1670" height="186" alt="image" src="https://github.com/user-attachments/assets/864c61b9-3ab3-4d1e-b193-63b9d6f40e8c" />

I recalled...there was a warning suggesting to pull it out. 

I didn't take a deep look. This seems a warning option could be modified. let me put it back.